haystack/components/eval/statistical_evaluator.py

```diff
@@ -0,0 +1,131 @@
+import collections
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from numpy import array as np_array
+from numpy import mean as np_mean
+
+from haystack import default_from_dict, default_to_dict
+from haystack.core.component import component
+
+from .preprocess import _preprocess_text
+
+
+@component
+class StatisticalEvaluator:
+    """
+    StatisticalEvaluator is a component that evaluates the performance of a model based on statistical metrics.
+    It's usually used in QA and Retrieval Augmented Generation (RAG) pipelines to evaluate the quality of the generated answers.
+
+    The supported metrics are:
+    - F1: Measures word overlap between predictions and labels.
+    - Exact Match: Measures the proportion of cases where prediction is identical to the expected label.
+    """
+
+    class Metric(Enum):
+        """
+        Supported metrics
+        """
+
+        F1 = "F1"
+        EM = "Exact Match"
+
+    def __init__(
+        self,
+        labels: List[str],
```

Comment:
This should be an input instead. (cf https://github.com/deepset-ai/haystack/pull/6980/files#r1489849342)